src/key_io.cpp

```diff
@@ -241,14 +243,29 @@ CTxDestination DecodeDestination(const std::string& str, const CChainParams& par
     data.clear();
     auto blech = blech32::Decode(str);
     const std::string& bl_hrp = for_parent ? params.ParentBlech32HRP() : params.Blech32HRP();
-    if (blech.second.size() > 0 && blech.first == bl_hrp) {
+    if (blech.data.size() > 0) {
```

Comment:
Lost the check here for `blech.encoding == blech32::Encoding::BECH32 || blech.encoding == blech32::Encoding::BECH32M` ?

Reply:
(Added on the bech32 side but not correspondingly here.)

Reply:
Yep, good catch. I think I confused myself here thinking that the extra address formats in Elements meant that I didn't need/want this check. But we do, the check is there to make sure that `blech32::Decode` doesn't return an error.